python/tests/test_dataset.py

```diff
@@ -801,3 +801,33 @@ def test_iteration():
     expected = ['a', 'b']
     for k in d:
         assert k in expected
+
+
+def test_dataset_html():
```

Comment:
This is not actually testing `_repr_html_`, but `__repr__`. Apart from changing the name for this newly added test, does your fix in `string.cpp` also work for `_repr_html_`, or does it need separate work?

Reply:
_repr_html_ works here as well. Added relevant tests.